HC-355 miscellaneous improvements to notebook_wrapper_pge

DustinKLo opened the pull request:
related ticket: https://hysds-core.atlassian.net/browse/HC-355

DustinKLo committed:
added PyYAML and Jinja2 dependencies

added Dockerfile generation through the use of a settings.yml and jinja templating
added Dockerfile.template

DustinKLo committed:
fixed readme

DustinKLo committed:
changed dockerfile update command

update README.md and templates/README.md

pymonger reviewed (dismissed):
LGTM

alexdunncs committed:
Merge branch 'develop' into HC-355

alexdunncs commented:
@DustinKLo have resolved merge conflict - seemed trivial, but take a look yourself to confirm?

DustinKLo commented:
thanks @alexdunncs
ill take a look at it later, there's still some stuff that i want to change on this PR

DustinKLo committed:
fixed bug where pele_setup.ipynb wasnt copied to created project

DustinKLo commented on notebook_pge_wrapper/cli.py:
```diff
@@ -88,7 +127,38 @@ def create(project):
         fout.write(templated_submit_job_content)
 
     # create pele_setup notebook
-    copyfile(pele_setup_notebook_file, os.path.join(project_root, __PELE_SETUP_NOTEBOOK_FILE))
```
fixed bug where `pele_setup.ipynb` wasn't being copied to newly created project

DustinKLo committed:
using latest tag in docker image

DustinKLo committed:
correct artifactory image endpoint (i think)

DustinKLo committed:
un-did artifactory image

pymonger commented on templates/Dockerfile.template:
```diff
@@ -0,0 +1,24 @@
+FROM {{ base_image }}
+
+ENV HOME=/home/ops \
```
```suggestion
ENV HOME=/home/{{ user }} \
```

pymonger commented on templates/Dockerfile.template:
```diff
@@ -0,0 +1,24 @@
+USER root
+RUN $CONDA_DIR/bin/conda install -y nb_conda_kernels
+
+USER ops
```
```suggestion
USER {{ user }}
```

pymonger commented on templates/Dockerfile.template:
```diff
@@ -0,0 +1,24 @@
+RUN $CONDA_DIR/bin/conda install -y nb_conda_kernels
+
+USER ops
+RUN $HOME/verdi/bin/pip install 'ipykernel<5.0.0'
```
Should be moved above under line 7 and modified to:
```suggestion
RUN $CONDA_DIR/bin/pip install 'ipykernel<5.0.0'
```

pymonger commented on templates/Dockerfile.template:
```diff
@@ -0,0 +1,24 @@
+USER ops
+RUN $HOME/verdi/bin/pip install 'ipykernel<5.0.0'
+
+RUN cd $HOME \
```
Should comment out these since notebook_pge_wrapper should be installed already. Add comment that user can uncomment if they want the latest installed.

pymonger commented on templates/Dockerfile.template:
```diff
@@ -0,0 +1,24 @@
+################################################
+COPY . $HOME/{{ project }}
+################################################
+
```
Need to add the running of requirement.ipynb here.

DustinKLo replied:
i'll handle it in a different PR b/c its in a [separate ticket](https://hysds-core.atlassian.net/browse/HC-353)
but will do

DustinKLo committed:
check for existence for settings.yml in ~/.config/

if doesnt exist will copy from templates/
will now read the settings.yml from ~/.config/notebook-pge-wrapper and propogate the Dockerfile values

cleaned up Dockerfile.template

DustinKLo commented:
@pymonger @jimmie made the changes discussed yesterday

`notebook-pge-wrapper create <project>` will now check is a `settings.yml` file exists in `~/.config/notebook-pge-wrapper`, if it doesn't then it'll create the directory and copy it over from `templates/`
the cli sub-commands will then use `settings.yml` from the `.config` directory and propagate `Dockerfile` with its values (optional to specify your own settings file with the `--settings` flag)

can you test it on your end first? it works on mine i just wanted to know if its user friendly enough to merge

DustinKLo committed:
added extra info about settings.yml in README.md

DustinKLo committed:
moved settings.yml generation away from __init__

added --settings flag to "create" and "dockerfile" (optional) to propogate Dockerfile values

pymonger commented on templates/Dockerfile.template:
```diff
@@ -0,0 +1,22 @@
+# RUN cd $HOME \
+#     && git clone https://github.com/hysds/notebook_pge_wrapper.git \
+#     && cd notebook_pge_wrapper \
+#     && $HOME/verdi/bin/pip install -e .
```
```suggestion
#     && $CONDA_DIR/bin/pip install -e .
```

DustinKLo replied:
good catch 👍

pymonger reviewed (commented):
One last fix

DustinKLo committed:
fixed commented out pip install in Dockerfile.template

pymonger reviewed (approved):
LGTM
